### PR TITLE
[ESP32] Diagnostic logs provider delegate implementation in temperature-meausement-app

### PR DESCRIPTION
**Pull request by shubhamdp**
#### Change Overview
- Implemented diagnostics logs provider delegate in temperature-measurement-app
- Enable mandatory bits for diagnostics log cluster in temperature-measurement-app zap file
- Added a config option to use the BDX for diagnostics log transfer

#### Tests
- Tested manually using interactive mode in chip-tool

### CODE REVIEW

File: examples/temperature-measurement-app/esp32/README.md [hunk 1]
```diff
@@ -24,6 +24,36 @@ Usage:
   ./out/debug/chip-tool temperaturemeasurement read measured-value <NODE ID> 1
 ```
 
+## Additional details
+
+This example demonstrates the utilization of the diagnostics logs cluster to
```
Comment (dhrishi) [inline examples/temperature-measurement-app/esp32/README.md:29]: Nit: diagnostic logs cluster

File: examples/temperature-measurement-app/esp32/README.md [hunk 2]
```diff
@@ -24,6 +24,36 @@ Usage:
...
+#### To test the diagnostic logs cluster
+
+```
+# Commission the app
+chip-tool pairing ble-wifi 1 SSID PASSPHRASE 20202021 3840
+
+# Read end user support logs using response payload protocol
+chip-tool diagnosticlogs retrieve-logs-request 0 0 1 0
+
+# Read network diagnostic using bdx protocol
```
Comment (dhrishi) [inline examples/temperature-measurement-app/esp32/README.md:53]: bdx -> BDX?

File: examples/temperature-measurement-app/esp32/main/diagnostic_logs/end_user_support.log [hunk 3]
```diff
@@ -0,0 +1 @@
+// dummy end user support log file
```
Comment (dhrishi) [inline examples/temperature-measurement-app/esp32/main/diagnostic_logs/end_user_support.log:1]: Can we actually have some dummy data

File: examples/temperature-measurement-app/esp32/main/diagnostic_logs/network_diag.log [hunk 4]
```diff
@@ -0,0 +1 @@
+// dummy network diagnostic log file
```
Comment (dhrishi) [inline examples/temperature-measurement-app/esp32/main/diagnostic_logs/network_diag.log:1]: Here as well, dummy data?

**Review by dhrishi — APPROVED** [review]

**Comment by dhrishi** [comment]
@vivien-apple Can you please review?

**Review by woody-apple — APPROVED** [review]

**Pull request merged**